src/libmongoc/tests/test-mongoc-client.c

```diff
@@ -3135,8 +3134,8 @@ test_mongoc_handshake_pool (void)
    mongoc_client_t *client1;
    mongoc_client_t *client2;
    mongoc_client_pool_t *pool;
-   const char *const server_reply =
-      "{'ok': 1, 'ismaster': true, 'minWireVersion': 2, 'maxWireVersion': 5}";
+   const char *const server_reply = "{'ok': 1, 'isWritablePrimary': true, "
```

Comment:
Minor: Since `maxWireVersion` is mocked as 5, I suggest using the legacy response field to make it more accurate. 
```suggestion
   const char *const server_reply = "{'ok': 1, '" HANDSHAKE_RESPONSE_LEGACY_HELLO "': true, "
```

Edit: there are a lot of cases like this, where the field name does not match with the mocked wire version. I don't feel too strongly since I think it is a low risk that SDAM incorrectly skips `isWritablePrimary` and we don't catch it somewhere else.

Reply:
I changed this one and didn't see there were other ones affected as well. I was trying to reproduce the server logic, instead opting for consistent usage of `isWritablePrimary` and only checking for legacy fields in a few specific tests.